lib/user_search_string.rb

```diff
@@ -0,0 +1,6 @@
+class UserSearchString < String
```

Comment:
I know that it should work in this particular case with only a constructor, but I'd still prefer just a class method like `UserSearchString.from(search_names)` instead of inheritance of a core type defined in C. Just to prevent someone else coming along adding a method on this class, and then being thoroughly confused at UserSearchStrings returning Strings when calling eg gsub.